nav2_system_tests/src/localization/CMakeLists.txt

```diff
@@ -1,19 +1,19 @@
-include_directories( ${PROJECT_SOURCE_DIR}/src )
-
 ament_add_gtest_executable(test_localization_node
   test_localization_node.cpp
 )
 ament_target_dependencies(test_localization_node
   ${dependencies}
 )
 
+
 ament_add_test(test_localization
   GENERATE_RESULT_FOR_RETURN_CODE_ZERO
   COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/test_localization_launch.py"
-  WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+  #WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
   TIMEOUT 20
   ENV
-    TEST_LAUNCH_DIR=${TEST_LAUNCH_DIR}
+    TEST_MAP=${PROJECT_SOURCE_DIR}/maps/map_circular.yaml
     TEST_EXECUTABLE=$<TARGET_FILE:test_localization_node>
+    TEST_WORLD=${PROJECT_SOURCE_DIR}/worlds/turtlebot3_ros2_demo.world
+    GAZEBO_MODEL_PATH=$ENV{GAZEBO_MODEL_PATH}
```

Comment:
This is an external dependency to the test. It will fail unless the user knows to set this before running colcon. That's undesirable, IMHO. We should discuss alternatives.

Reply:
I did some testing and without the GAZEBO_MODEL_PATH set, laser scans never get published. I think this has to be set before the package is built or else the test will fail.

Reply:
I can't see a way to get this variable out of CMake, so I guess leaving it as is for now is the way to go. We just need to document in some way that the environment variable needs to be set before building.